### PR TITLE
Add a sanity check for OpenFOAM's `wmake` to make sure it finds the compiler commands

### PR DESCRIPTION
**Pull request by ocaisa**

### CODE REVIEW

File: easybuild/easyblocks/o/openfoam.py [hunk 1]
```diff
@@ -486,6 +486,11 @@ def sanity_check_step(self):
             test_foammonitor = "! foamMonitor -h 2>&1 | grep 'not installed'"
             custom_commands.append(' && '.join([load_openfoam_env, test_foammonitor]))
 
+        if self.is_dot_com and self.looseversion >= LooseVersion("2012"):
```
Comment (ocaisa) [inline easybuild/easyblocks/o/openfoam.py:489]: This should always have worked, but I could be conservative and only actually check for later versions of OpenFOAM.com

**Comment by ocaisa** [comment]
@boegelbot please test @ generoso
EB_ARGS="OpenFOAM-v2112-foss-2022a.eb OpenFOAM-v2206-foss-2022a.eb"

**Comment by boegel** [comment]
Also tested with `OpenFOAM-v2012-foss-2020a.eb` and `OpenFOAM-v2106-foss-2021a.eb` using `--sanity-check-only`, works as intended.

**Comment by boegel** [comment]
Test report by @boegel
#### Overview of tested easyconfigs (in order)
 * **FAIL (build issue)** _OpenFOAM-v2306-foss-2022b.eb_ (partial log available at https://gist.github.com/boegel/1961ba81061d2779008ae6d406aa6bfa)
 * **FAIL (build issue)** _OpenFOAM-v2312-foss-2023a.eb_ (partial log available at https://gist.github.com/boegel/a63b64b502cbab6735ce253d09caa71c)
 * **SUCCESS** _OpenFOAM-10-foss-2023a.eb_ 
 * **SUCCESS** _OpenFOAM-11-foss-2023a.eb_ 

Build succeeded for 2 out of 4 (4 easyconfigs in total)
node3120.skitty.os - Linux RHEL 8.8, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/boegel/aab8ac20f8ac23230e1656bec04f8347 for a full test report.

**Comment by ocaisa** [comment]
@boegel Those builds failed as they are the ones fixed in https://github.com/easybuilders/easybuild-easyconfigs/pull/20517

**Comment by ocaisa** [comment]
(which was the motivation for this PR)

**Review by boegel — APPROVED** [review]

**Pull request merged**